enhancements/mcd-reboot-dictionary.md

```diff
@@ -0,0 +1,215 @@
+
+- Should the parts of the configuration that do and don't require reboot be
+  part of user facing documentation?
+
+
+## Summary
+
+A node's configuration is managed by the MCO.  When the configuation is changed
+by the system or an admin, the MCO and MCD write a new version of the file to
+disk and reboot the system to ensure it is applied.
+
+This proposal is to create a dictionary that defines a whitelist of file paths
+for which alternative application strategies should be used.  
+
+
+## Motivation
+
+Always rebooting a node is reasonable for cloud based systems, since they can
+be rebooted very quickly.  Additionally, it is the only way for kernel
+parameter changes to take effect.
+
+However on baremetal systems, rebooting a node takes significantly longer (in
+the order of minutes) and the machine configuration includes files that need
+either no action to take effect, or can be more quickly applied by restarting a
+systemd service.
+
+### Goals
+
+- Creation of a minimally invasive solution that can be delivered in the 
+  short-term, allowing time for a more comprehensive approach to be developed
+- A system defined whitelist of file paths that do not trigger a reboot
+- Changes to systemd service configurations are applied by restarting the service
+- Specific files can be updated with no action (eg. ssh keys)
+- Any file not covered by the whitelist triggers a reboot
+
+### Non-Goals
+
+- Allowing the admin to specify additional paths or strategies
+- Allowing different behaviour based on which option or part of the file has
+  been updated
+
+## Proposal
+
+The proposal calls for a whitelist that MCD uses to decide which action to take
+after updating a configuration file for the MCO.  To prevent the admin from
+tampering with the whitelist, it will be included as part of the MCD image as a
+static file in yaml format.
+
+The information could be baked into the the MCD itself, but assuming we treat it
+as data, the file will be a list of entries containing:
+
+- filename
+- path
+- action to perform
+- action specific data (eg. the service name, timeout)
+
+After the MCD writes out a configuation change, it will consult the whitelist
+before deciding if a reboot is required. If the filename is present, has a
+valid action and any associated data, the action will be performed.  Otherwise,
+a reboot will be performed as in prior 4.x versions.
+
+If the action results in an error, or takes too long, a reboot will be
+performed as in prior 4.x versions.
+
+
+### User Stories [optional]
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+
+As a telco admin, I want the system to use the least invasive method for
+applying configuration file updates, to reduce periods of degraded system and
+application availability.
```

Comment:
hmm personally to me this is more like motivation and not user story.. a user story might be

- I want to update the SSH keys without reboot

- I want to update the kubelet config file without reboot, this requires no systemd unit restart

- I want to update the CRI-O configuration without reboot, this requires systemd unit restart

- I want to force reboot for network configuration update..

etc...

Reply:
I was trying to be generic, but it probably makes more sense to be explicit.
I'll update